modules/oe_content_event/src/EventNodeWrapper.php

```diff
@@ -146,4 +146,70 @@ public function isRegistrationPeriodOver(\DateTimeInterface $datetime): bool {
     return $this->hasRegistrationDates() && $datetime >= $this->getRegistrationEndDate()->getPhpDateTime();
   }
 
+  /**
+   * {@inheritdoc}
+   */
+  public function hasOnlineDates(): bool {
+    return !$this->entity->get('oe_event_online_dates')->isEmpty();
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function isOnlinePeriodYetToCome(\DateTimeInterface $datetime): bool {
+    if (!$this->hasOnlineDates()) {
+      return FALSE;
+    }
+
+    return $this->hasOnlineDates() && $datetime < $this->getOnlineStartDate()->getPhpDateTime();
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function isOnlinePeriodActive(\DateTimeInterface $datetime): bool {
+    return $this->hasOnlineDates()
+      && $this->getOnlineStartDate()->getPhpDateTime() <= $datetime
+      && $datetime < $this->getOnlineEndDate()->getPhpDateTime();
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function isOnlinePeriodOver(\DateTimeInterface $datetime): bool {
+    if (!$this->hasOnlineDates()) {
+      return TRUE;
```

Comment:
This is a bit misleadin, if there are no online dates its neither over nor it has not started. Lets follow the same logic as in isRegistrationPeriodOver.